File: 2390.cpp
#include<iostream>
#include<string>
#include<algorithm>

using namespace std;

class Solution {
public:
    string removeStars(string s) {
        string ans;
        for(char c : s){
            if(c == '*'){
                if(!ans.empty()) ans.pop_back();
            }
            else ans += c;
        }
        return ans;
    }
};

int main() {
    Solution s;
    string str;
    cin >> str;
    cout << s.removeStars(str) << endl;
    return 0;
}

File: 2390.py
class Solution:
    def removeStars(self, s: str) -> str:
        ans = []
        for c in s:
            if c == '*':
                if len(ans) > 0:
                    ans.pop()
            else:
                ans.append(c)
        return ''.join(ans)
    
if __name__ == '__name__':
    print(Solution().removeStars(input()))